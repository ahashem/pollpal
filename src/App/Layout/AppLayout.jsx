import React, { Component } from 'react';
import { Layout } from 'antd';
import AppHeader from './AppHeader';
import AppContent from './AppContent';

/**
 * Application layout using AntD layout components
 */
class AppLayout extends Component {
  state = {
    collapsed: false,
  };

  /**
   * Toggle Side menu view
   */
  onCollapse = (collapsed) => {
    this.setState({ collapsed });
  };

  render() {
    const renderHeader = false;

    return (
      <Layout>
        <Layout>
          {renderHeader ? <AppHeader /> : ''}
          <AppContent>
            {this.props.children}
          </AppContent>
        </Layout>
      </Layout>
    );
  }
}

export default AppLayout;
