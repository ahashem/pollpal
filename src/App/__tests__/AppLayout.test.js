import React from 'react';
import { shallow } from 'enzyme/build';
import AppLayout from '../Layout/AppLayout';

describe('<AppLayout /> component', () => {
  it('should render children', () => {
    const mockChildren = [];

    const rendered = shallow(
      <AppLayout>
        {mockChildren}
      </AppLayout>
    );

    expect(rendered).toMatchSnapshot();
  });
});
