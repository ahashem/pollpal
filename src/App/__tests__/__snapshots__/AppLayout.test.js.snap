// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppLayout /> component should render children 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppLayout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Adapter>
        
        <AppContent />
      </Adapter>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          "",
          <AppContent />,
        ],
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Adapter>
          
          <AppContent />
        </Adapter>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "",
            <AppContent />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [],
            },
            "ref": null,
            "rendered": Array [],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
